### PR TITLE
Sync: Test Script

### PR DESCRIPTION
**Pull request by malmstein**
Task/Issue URL: https://app.asana.com/0/488551667048375/1205651033893049/f

### Description
Adds a new workflow for Sync E2E critical path tests

### Steps to test this PR
https://github.com/duckduckgo/Android/actions/runs/6533958436 for existing tests
Maestro test locally for syncCriticalPathTests tag

### CODE REVIEW

**Comment by malmstein** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3621** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3621" target="_blank">Graphite</a> 
    * **PR #3622** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3622" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/3622?utm_source=stack-comment).

File: .github/workflows/sync-critical-path.yml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+          distribution: 'adopt'
+
+      - name: Create folder
+        if: always()
+        run: mkdir apk
+
+      - name: Decode keys
+        uses: davidSchuppa/base64Secret-toFile-action@v2
+        with:
+          secret: ${{ secrets.FAKE_RELEASE_PROPERTIES }}
```
Comment (cmonfortep) [inline .github/workflows/sync-critical-path.yml:36]: Should we update to not Fake? Or is this the actual property we want?

File: .github/workflows/sync-critical-path.yml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+          fileName: ddg_android_build.properties
+          destination-path: $HOME/jenkins_static/com.duckduckgo.mobile.android/
+
+      - name: Decode key file
+        uses: davidSchuppa/base64Secret-toFile-action@v2
+        with:
+          secret: ${{ secrets.FAKE_RELEASE_KEY }}
```
Comment (cmonfortep) [inline .github/workflows/sync-critical-path.yml:43]: same question.

File: .github/workflows/sync-critical-path.yml [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+      - name: Decode keys
+        uses: davidSchuppa/base64Secret-toFile-action@v2
+        with:
+          secret: ${{ secrets.FAKE_RELEASE_PROPERTIES }}
+          fileName: ddg_android_build.properties
+          destination-path: $HOME/jenkins_static/com.duckduckgo.mobile.android/
+
+      - name: Decode key file
+        uses: davidSchuppa/base64Secret-toFile-action@v2
```
Comment (jpbochi) [inline .github/workflows/sync-critical-path.yml:41]: who's david schuppa?

Comment (malmstein) [reply .github/workflows/sync-critical-path.yml:41]: it’s an action we use to be able to sign apks. we are not related :trollface:

**Pull request merged**